mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/OtpTextField.kt

```diff
@@ -0,0 +1,120 @@
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun OtpTextField(
+    modifier: Modifier = Modifier,
+    realOtp: String = "",
+    otpCount: Int = 4,
+    onOtpTextCorrectlyEntered : () -> Unit
+) {
+    var otpText by remember { mutableStateOf("") }
+    var isError by remember { mutableStateOf(false) }
+   Column(
+         modifier = modifier
+             .fillMaxWidth()
+             .padding(16.dp),
+       horizontalAlignment = Alignment.CenterHorizontally
+   ) {
+
+       BasicTextField(
+           modifier = modifier,
+           value = TextFieldValue(otpText, selection = TextRange(otpText.length)),
+           onValueChange = {
+               otpText = it.text
+                isError = false
+           },
+           keyboardActions = KeyboardActions(onDone =
+           {
+              if(otpText != realOtp) {
+                  isError = true
+              } else {
+                  onOtpTextCorrectlyEntered.invoke()
+              }
+               println("OTP: $otpText and $isError")
+           }),
+           keyboardOptions = KeyboardOptions.Default.copy(
+               keyboardType = KeyboardType.Number,
+               imeAction = ImeAction.Done
+           ),
+           decorationBox = {
+               Row(horizontalArrangement = Arrangement.Center) {
+                   repeat(otpCount) { index ->
+                       CharView(
+                           index = index,
+                           text = otpText
+                       )
+                       Spacer(modifier = Modifier.width(8.dp))
+                   }
+               }
+           }
+       )
+       if (isError) {
+           // display erro message in text
+           Text(
+               text = "Invalid OTP",
+               style = MaterialTheme.typography.body1,
+               color = Color.Red,
+               textAlign = TextAlign.Center,
+               modifier = Modifier.padding(top = 8.dp)
+           )
+       }
+   }
+}
+
+@Composable
+fun CharView(
+    index: Int,
+    text: String
+) {
+    val isFocused = text.length == index
+    val char = when {
+        index == text.length -> "_"
+        index > text.length -> "_"
+        else -> text[index].toString()
+    }
+    Text(
+        modifier = Modifier
+            .width(40.dp)
+            .border(
+                1.dp, when {
+                    isFocused -> Color.DarkGray
+                    else -> Color.LightGray
+                }, RoundedCornerShape(8.dp)
+            )
+            .wrapContentHeight(align = Alignment.CenterVertically),
+        text = char,
+        style = MaterialTheme.typography.h4,
+        color = if (isFocused) {
+            Color.LightGray
+        } else {
+            Color.DarkGray
+        },
+        textAlign = TextAlign.Center
+    )
+}
```

Comment:
I don't see any preview function, Please add preview function for all the composable functions

Reply:
preview for every composable done